node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/task/util/file/FileFinder.java

```diff
@@ -110,17 +110,19 @@ public FileFinder maxDepth(int maxDepth) {
      * @return true iff anything was matched and deleted
      */
     public boolean deleteRecursively(TaskContext context) {
+        final int limit = 20;
+        int[] numDeleted = { 0 };
```

Comment:
Why not use a for loop on line 118 instead to avoid this hack?

Reply:
At its core, this class only has a way a method to run a lambda on a matching file via `java.nio.file.Files::walkFileTree`, I don't see any simple way to create an `Iterable` out of that?

Reply:
Ah, right. Didn't look at the context. Then I suggest adding a comment, or using MutableInteger.